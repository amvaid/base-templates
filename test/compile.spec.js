const path = require('path');
const jade = require('jade-legacy');
const compile = require('../packages/oc-template-jade-compiler/compile');

describe('compile method', () => {
  describe('when invoking the method', () => {
    const template = 'Hello';
    const viewPath = path.resolve(__dirname);

    test('should correctly invoke jade compileClient method', (done) => {
      compile({ template, viewPath }, () => {
        expect(jade.compileClient).toBeCalledWith(template, {
          compileDebug: false,
          filename: path.resolve(__dirname),
          name: 't'
        });
        done();
      });
    });
  });
});
