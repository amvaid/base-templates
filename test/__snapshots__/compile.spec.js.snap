// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compile method when invoking the method should correctly invoke jade compileClient method 1`] = `null`;

exports[`compile method when invoking the method should correctly invoke jade compileClient method 2`] = `"Hello"`;

exports[`compile method when invoking the method should correctly throw 1`] = `[Error: Whoops!]`;

exports[`compile method when invoking the method should correctly throw 2`] = `undefined`;
